yt_dlp/extractor/vrt.py

```diff
@@ -231,100 +293,76 @@ class VrtNUIE(VRTBaseIE):
-                    }))
-            except ExtractorError as e:
-                if isinstance(e.cause, HTTPError) and e.cause.status == 401:
-                    retry.error = e
-                    continue
-                raise
+        self._request_webpage('https://www.vrt.be/vrtnu/sso/login', None, note='Getting session cookies', errnote='Failed to get session cookies')
 
+        self._download_json(
+            'https://login.vrt.be/perform_login', None, data=json.dumps({
+                "loginID": username,
+                "password": password,
+                "clientId": "vrtnu-site"
+            }).encode(), headers={
+                'Content-Type': 'application/json',
+                'Oidcxsrf': self._get_cookies('https://login.vrt.be').get('OIDCXSRF').value,
+            }, note='Logging in', errnote='Login failed')
         self._authenticated = True
+        return
 
     def _real_extract(self, url):
         display_id = self._match_id(url)
         parsed_url = urllib.parse.urlparse(url)
-        details = self._download_json(
-            f'{parsed_url.scheme}://{parsed_url.netloc}{parsed_url.path.rstrip("/")}.model.json',
-            display_id, 'Downloading asset JSON', 'Unable to download asset JSON')['details']
-
-        watch_info = traverse_obj(details, (
-            'actions', lambda _, v: v['type'] == 'watch-episode', {dict}), get_all=False) or {}
-        video_id = join_nonempty(
-            'episodePublicationId', 'episodeVideoId', delim='$', from_dict=watch_info)
-        if '$' not in video_id:
-            raise ExtractorError('Unable to extract video ID')
-
-        vrtnutoken = self._download_json(
-            'https://token.vrt.be/refreshtoken', video_id, note='Retrieving vrtnutoken',
-            errnote='Token refresh failed')['vrtnutoken'] if self._authenticated else None
-
-        video_info = self._call_api(video_id, 'vrtnu-web@PROD', vrtnutoken)
-
-        if 'title' not in video_info:
-            code = video_info.get('code')
-            if code in ('AUTHENTICATION_REQUIRED', 'CONTENT_IS_AGE_RESTRICTED'):
-                self.raise_login_required(code, method='password')
-            elif code in ('INVALID_LOCATION', 'CONTENT_AVAILABLE_ONLY_IN_BE'):
-                self.raise_geo_restricted(countries=['BE'])
-            elif code == 'CONTENT_AVAILABLE_ONLY_FOR_BE_RESIDENTS_AND_EXPATS':
-                if not self._authenticated:
-                    self.raise_login_required(code, method='password')
-                self.raise_geo_restricted(countries=['BE'])
-            raise ExtractorError(code, expected=True)
-
-        formats, subtitles = self._extract_formats_and_subtitles(video_info, video_id)
+
+        # 1. Obtain/refresh 'vrtnu-site_profile' tokens
+        self._request_webpage('https://www.vrt.be/vrtnu/sso/login', None, note='Getting tokens', errnote='Failed to get tokens')
+
+        # 2. Perform GraphQL query to obtain video metadata
```

Comment:
I think you can leave out the comments, most of this is pretty self-explanatory.

Reply:
As a general rule of thumb, comments that explain "what" are pointless as opposed to comments that explain "why"